src/gui/TimeLineWidget.cpp

```diff
@@ -224,55 +215,67 @@ void TimeLineWidget::paintEvent( QPaintEvent * )
-	p.setOpacity( 1.0 );
 
+	// Draw the loop rectangle
+	int const loopRectMargin = 1;
+	int const loopRectHeight = this->height() - 2 * loopRectMargin;
+	int const loopStart = markerX( loopBegin() ) + 8;
+	int const loopEndR = markerX( loopEnd() ) + 9;
+	int const loopRectWidth = loopEndR - loopStart;
 
-	tact_t tact_num = m_begin.getTact();
-	int x = m_xOffset + s_posMarkerPixmap->width() / 2 -
-			( ( static_cast<int>( m_begin * m_ppt ) / MidiTime::ticksPerTact() ) % static_cast<int>( m_ppt ) );
+	bool const loopPointsActive = loopPointsEnabled();
 
-	QColor lineColor( 192, 192, 192 );
-	QColor tactColor( lineColor.darker( 120 ) );
+	// Draw the main rectangle (inner fill only)
+	QRect outerRectangle( loopStart, loopRectMargin, loopRectWidth - 1, loopRectHeight - 1 );
+	p.fillRect( outerRectangle, loopPointsActive ? getActiveLoopBrush() : getInactiveLoopBrush());
 
-	// Set font to half of the widgets size (in pixels)
+	// Draw the bar lines and numbers
+	// Activate hinting on the font
 	QFont font = p.font();
-	font.setPixelSize( this->height() * 0.5 );
-	p.setFont( font );
+	font.setHintingPreference( QFont::PreferFullHinting );
+	p.setFont(font);
+	int const fontAscent = p.fontMetrics().ascent();
+	int const fontHeight = p.fontMetrics().height();
+
+	QColor const & barLineColor = getBarLineColor();
+	QColor const & barNumberColor = getBarNumberColor();
+
+	tact_t barNumber = m_begin.getTact();
+	int const x = m_xOffset + s_posMarkerPixmap->width() / 2 -
+			( ( static_cast<int>( m_begin * m_ppt ) / MidiTime::ticksPerTact() ) % static_cast<int>( m_ppt ) );
 
 	for( int i = 0; x + i * m_ppt < width(); ++i )
 	{
-		const int cx = x + qRound( i * m_ppt );
-		p.setPen( lineColor );
-		p.drawLine( cx, 5, cx, height() - 6 );
-		++tact_num;
-		if( ( tact_num - 1 ) %
+		++barNumber;
+		if( ( barNumber - 1 ) %
 			qMax( 1, qRound( 1.0f / 3.0f *
 				MidiTime::ticksPerTact() / m_ppt ) ) == 0 )
 		{
-			const QString s = QString::number( tact_num );
-			p.setPen( tactColor );
-			p.drawText( cx + qRound( ( m_ppt - p.fontMetrics().
-							width( s ) ) / 2 ),
-				height() - p.fontMetrics().ascent() / 2, s );
+			const int cx = x + qRound( i * m_ppt );
+			p.setPen( barLineColor );
+			p.drawLine( cx, 5, cx, height() - 6 );
+
+			const QString s = QString::number( barNumber );
+			p.setPen( barNumberColor );
+			p.drawText( cx + 5, ((height() - fontHeight) / 2) + fontAscent, s );
 		}
 	}
 
-	p.setOpacity( 0.6 );
-	p.drawPixmap( m_posMarkerX, height() - s_posMarkerPixmap->height(),
-							*s_posMarkerPixmap );
+	// Draw the main rectangle (outer border)
+	p.setPen( loopPointsActive ? getActiveLoopColor() : getInactiveLoopColor() );
+	p.setBrush( Qt::NoBrush );
+	p.drawRect( outerRectangle );
+
+	// Draw the inner border outline (no fill)
+	QRect innerRectangle = outerRectangle.adjusted( 1, 1, -1, -1 );
+	p.setPen( loopPointsActive ? getActiveLoopInnerColor() : getInactiveLoopInnerColor() );
+	p.setBrush(Qt::NoBrush);
```

Comment:
I'd add spaces in the parentheses here, so its consistent with the rest of the file.